src/caustics/sims/lens_source.py

```diff
@@ -91,18 +120,21 @@ def __init__(
             Optional[Source],
             "caustics light object which defines the lensing object's light",
         ] = None,
-        psf: Annotated[Optional[Tensor], "An image to convolve with the scene"] = None,
         pixels_y: Annotated[
             Optional[int], "number of pixels on the y-axis for the sampling grid"
         ] = None,
         upsample_factor: Annotated[int, "Amount of upsampling to model the image"] = 1,
-        psf_pad: Annotated[bool, "Flag to apply padding to psf"] = True,
+        quad_level: Annotated[Optional[int], "sub pixel integration resolution"] = None,
         psf_mode: Annotated[
             Literal["fft", "conv2d"], "Mode for convolving psf"
         ] = "fft",
+        psf_shape: Annotated[Optional[tuple[int, ...]], "The shape of the psf"] = None,
         z_s: Annotated[
             Optional[Union[Tensor, float]], "Redshift of the source", True
         ] = None,
+        psf: Annotated[
```

Comment:
Would make it None, easier to handle as a default.

Reply:
None as a default means dynamic, so user specified. I wanted the default to not require a PSF. also there is a natural "null psf" which is the 1x1 kernel since that must always be the identity because `psf = psf / psf.sum()`.